### PR TITLE
Task migration from using Resources to using workspaces

### PR DESCRIPTION
**Pull request by Preeticp**
This PR fixes: 

 - https://issues.redhat.com/browse/RHDEVDOCS-2081
 - https://issues.redhat.com/browse/RHDEVDOCS-1693
 - https://issues.redhat.com/browse/RHDEVDOCS-2109

 The YAML files in the creating workflow and Understanding conceptual docs have been updated to reflect the migration from Resources to Workspaces. Notes for migration added. 
New module for pvc volumes added.
This also fixes the bug of adding cloning the tutorial repo to the Prerequisites in the Creating flow.

- This has been reviewed by the QE 
- This needs to be merged to 4.5 and 4.6

### CODE REVIEW

**Comment by Preeticp** [comment]
@ppitonak @khrm @piyush-garg request a review please.

**Review by ppitonak — CHANGES_REQUESTED** [review]

**Review by ppitonak — CHANGES_REQUESTED** [review]
In section Creating Webhooks, we instruct user to create a webhook in frontend repo `Configure Webhooks manually on the front-end repository:` but later we commit to api repo.

File: modules/op-running-a-pipeline.adoc
Comment (ppitonak) [reply modules/op-running-a-pipeline.adoc:16]: example https://github.com/openshift/pipelines-tutorial/blob/release-tech-preview-2/01_pipeline/03_persistent_volume_claim.yaml

File: modules/op-assembling-a-pipeline.adoc
Comment (khrm) [reply modules/op-assembling-a-pipeline.adoc:9]: Yes, they referred to Pipelineresources. If you want to define the order for input and output, we need `from` clause.

File: pipelines/creating-applications-with-cicd-pipelines.adoc
Comment (Preeticp) [reply pipelines/creating-applications-with-cicd-pipelines.adoc:23]: I had a conflict and looks like that's when this was introduced!

File: pipelines/creating-applications-with-cicd-pipelines.adoc
Comment (Preeticp) [reply pipelines/creating-applications-with-cicd-pipelines.adoc:36]: Will do

Comment (Preeticp) [reply pipelines/creating-applications-with-cicd-pipelines.adoc:36]: Since it was not yet deprecated, had commented it out.

File: modules/op-about-triggers.adoc
Comment (ppitonak) [reply modules/op-about-triggers.adoc:106]: this was marked as resolved but it didn't change

File: modules/op-running-a-pipeline.adoc
Comment (Preeticp) [reply modules/op-running-a-pipeline.adoc:16]: Created another section

**Comment by Preeticp** [comment]
Good catch fixed the commit section in Triggering a PipelineRun

**Comment by Preeticp** [comment]
@ppitonak I have added a section for the PVC volumes and added notes wrt migration from resources to workspaces in 2 places as discussed. Request you to review please.

File: modules/op-specifying-pvc-as-volumesource-in-workspaces.adoc
Comment (ppitonak) [inline modules/op-specifying-pvc-as-volumesource-in-workspaces.adoc:40]: This needs to be updated to tech-preview branch... https://raw.githubusercontent.com/openshift/pipelines-tutorial/release-tech-preview-2/01_pipeline/03_persistent_volume_claim.yaml

**Review by ppitonak — CHANGES_REQUESTED** [review]

**Review by ppitonak — APPROVED** [review]

**Comment by ppitonak** [comment]
LGTM

File: modules/op-assembling-a-pipeline.adoc
Comment (bobfuru) [inline modules/op-assembling-a-pipeline.adoc:110]: - If you are calling attention to this chunk, it should be wrapped in a note/admonition style.
- The language is unclear and I'm not sure what I should be noticing. Suggest switching to more active voice, something like, "The Pipeline definition abstracts away the details of the Git source repository and image registries. These details are added as `params` when execution triggers a Pipeline."

File: modules/op-release-notes-1-1.adoc
Comment (bobfuru) [inline modules/op-release-notes-1-1.adoc:21]: A suggested revision for readability and style:

> Workspaces can now be used instead of PipelineResources. Use of Workspaces is recommended in OpenShift Pipelines, as PipelineResources are difficult to debug, limited in scope, and can make Tasks less reusable. For more details on Workspaces, see **Understanding OpenShift Pipelines** (with link to assembly).

(For the last part, I don't think the reader will be able to understand what is meant by "concepts section.")

File: modules/op-specifying-pvc-as-volumesource-in-workspaces.adoc
Comment (bobfuru) [inline modules/op-specifying-pvc-as-volumesource-in-workspaces.adoc:8]: s/Workspaces help share data among Tasks and /Workspaces help Tasks share data, and/

File: modules/op-specifying-pvc-as-volumesource-in-workspaces.adoc
Comment (bobfuru) [inline modules/op-specifying-pvc-as-volumesource-in-workspaces.adoc:10]: I'm glad to see "workspace" and "pipeline" in lowercase here, though it doesn't follow the conventions of capitalizing objects elsewhere in the assembly. I suggest capitalizing here for consistency (at least until the business is convinced otherwise 😉 ).

File: modules/op-specifying-pvc-as-volumesource-in-workspaces.adoc
Comment (bobfuru) [inline modules/op-specifying-pvc-as-volumesource-in-workspaces.adoc:15]: Suggest: "Copy and save the contents..."
Alternately, you might not want to instruct the user to copy this if it is sample data. So you might say something like, "Create a PersistentVolumeClaim YAML file similar to the following example:"

File: modules/op-specifying-pvc-as-volumesource-in-workspaces.adoc
Comment (bobfuru) [inline modules/op-specifying-pvc-as-volumesource-in-workspaces.adoc:31]: For clarity:
s/. Create the PersistentVolumeClaim:/. Create the PersistentVolumeClaim, specifying the file you just created:/

File: modules/op-triggering-a-pipelinerun.adoc
Comment (bobfuru) [inline modules/op-triggering-a-pipelinerun.adoc:10]: I know this assembly uses the convention of, "In this section, you will..." but it doesn't seem to align with our overall docs practice of using a more active voice (as opposed to "you will").

If you're so inclined, I'd suggest revising to something like, 

> The following example describes how to push an empty commit to the front-end `vote-ui` repository, which triggers the PipelineRun.

File: pipelines/creating-applications-with-cicd-pipelines.adoc
Comment (bobfuru) [inline pipelines/creating-applications-with-cicd-pipelines.adoc:14]: Here pipeline is lowercased in "pipeline execution."

File: pipelines/creating-applications-with-cicd-pipelines.adoc
Comment (bobfuru) [inline pipelines/creating-applications-with-cicd-pipelines.adoc:21]: There should also be a comma after interface in these examples ("...front-end interface, `vote-ui`, with...").

File: pipelines/creating-applications-with-cicd-pipelines.adoc
Comment (bobfuru) [inline pipelines/creating-applications-with-cicd-pipelines.adoc:29]: s/GitHub ID and have/GitHub ID, and you have/

File: pipelines/creating-applications-with-cicd-pipelines.adoc
Comment (bobfuru) [inline pipelines/creating-applications-with-cicd-pipelines.adoc:50]: Thanks for updating this. 😁

**Review by bobfuru — COMMENTED** [review]
Some general comments for readability and style. Also, could you please squash commits and remove WIP label when ready for merge? Thanks!

File: modules/op-release-notes-1-1.adoc
Comment (Preeticp) [reply modules/op-release-notes-1-1.adoc:21]: I would have loved to add a link but xrefs can not be added to modules, right?

File: modules/op-specifying-pvc-as-volumesource-in-workspaces.adoc
Comment (Preeticp) [reply modules/op-specifying-pvc-as-volumesource-in-workspaces.adoc:15]: For this entire workflow to work they need to use this particular YAML file hence instructions to copy. Otherwise I like the second option too.

**Comment by Preeticp** [comment]
@bobfuru Thank you. 
I have fixed all the comments except the ones which deal with code snippets. I will discuss the code changes with the devs and create an issue for them, I think they might have to resolve it in the Upstream too. Build/built was something on my mind too.
I also have a question wrt adding links to modules. 
PTAL and merge if this looks alright to you.

File: modules/op-release-notes-1-1.adoc
Comment (bobfuru) [reply modules/op-release-notes-1-1.adoc]: Oh, right. Sorry, didn't realize this was a module so ignore my link suggestion.

File: modules/op-about-workspace.adoc
Comment (bobfuru) [inline modules/op-about-workspace.adoc:31]: Commas in this sentence are a bit awkward; didn't catch it before. Suggestion:
The following example shows a code snippet of the `build-and-deploy` Pipeline, which declares a `shared-workspace` Workspace for the `build-image` and `apply-manifests` Tasks as defined in the Pipeline.

File: modules/op-about-workspace.adoc
Comment (bobfuru) [inline modules/op-about-workspace.adoc:73]: I would avoid the `-` in this sentence and say something like, "...the Pipeline Workspace named `shared-workspace`.

File: modules/op-release-notes-1-1.adoc
Comment (bobfuru) [inline modules/op-release-notes-1-1.adoc:21]: For the first sentence, I think you can remove "is now provided."

File: modules/op-specifying-pvc-as-volumesource-in-workspaces.adoc
Comment (bobfuru) [inline modules/op-specifying-pvc-as-volumesource-in-workspaces.adoc:31]: Remove extra colon

File: modules/op-specifying-pvc-as-volumesource-in-workspaces.adoc
Comment (bobfuru) [inline modules/op-specifying-pvc-as-volumesource-in-workspaces.adoc:37]: s/you can also execute/you can execute/

**Comment by Preeticp** [comment]
I have created a [Jira ](https://issues.redhat.com/browse/SRVKP-998) for the snippet changes and will send a follow up PR for it. IMHO that should not block this PR.

**Review by bobfuru — COMMENTED** [review]
Overall, looks good with a few more minor revisions needed before merging. Makes sense about the code snippet issue and spec descriptions. Also, can you please squash commits? Thanks!

File: modules/op-release-notes-1-1.adoc
Comment (Preeticp) [reply modules/op-release-notes-1-1.adoc:21]: Thank you for pointing that!! Not sure how I missed that!

**Comment by Preeticp** [comment]
Fixed comments and squashed @bobfuru Request to merge to 4.5 and 4.6

**Comment by bobfuru** [comment]
LGTM!

**Pull request merged**
